internal/reload/watcher.go

```diff
@@ -0,0 +1,128 @@
+package reload
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"os/exec"
+	"sync"
+
+	"github.com/fsnotify/fsnotify"
+)
+
+type Watcher struct {
+	clients map[chan string]bool
+	watcher *fsnotify.Watcher
+	mu      sync.Mutex
+}
+
+// NewWatcher creates a new watcher for the given path.
+// The watcher listens for file changes and broadcasts them to all connected clients.
+// Note: Usually you only ever have one client as this is intended for live-reloading
+// a web page in a development environment.
+func NewWatcher(path string) (*Watcher, error) {
+	watcher := &Watcher{
+		clients: make(map[chan string]bool),
+	}
+	w, err := fsnotify.NewWatcher()
+	if err != nil {
+		log.Printf("Failed to create watcher: %v", err)
+		return nil, err
+	}
+	defer w.Close()
```

Comment:
Just a heads up to future Jostein:

```go
thing, close := newThing()
defer close() // why we don't code late at night
go thing.needsToBeNotClosed()
return thing
```

Is not a smart way to keep things in the goroutine working.

(This comment was sponsored by the 7 gray strands of hair generated by line 32)